targets/TARGET_Cypress/TARGET_PSOC6/mbed_overrides.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * mbed Microcontroller Library
+ * Copyright (c) 2018-2019 Cypress Semiconductor Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "device.h"
+#include "psoc6_utils.h"
+#include "cycfg.h"
+
+#if defined(COMPONENT_SPM_MAILBOX)
+void mailbox_init(void);
+#endif
+
+/*******************************************************************************
+* Function Name: mbed_sdk_init
+****************************************************************************//**
+*
+* Mbed's post-memory-initialization function.
+* Used here to initialize common parts of the Cypress libraries.
+*
+*******************************************************************************/
+void mbed_sdk_init(void)
+{
+#if !defined(COMPONENT_SPM_MAILBOX)
+    /* Disable global interrupts */
+    __disable_irq();
```

Comment:
you should use rather critical section here

Reply:
Same for line 63 and 56

Reply:
As this was in the code previously, we could still fix it here.

Reply:
This is now fixed by https://github.com/ARMmbed/mbed-os/pull/10483/commits/737c98ff8fcae3cdfb7e231a8c719a793a21b8e7 - after further analysis, I can confirm entering critical section is not needed at all, init_cycfg_all is fine to execute with global interrupts enabled. I checked this functionally on CM4/CM0P PSA/non-PSA targets (actually there is no change in PSA case). On CM0+, global interrupts are enabled by ROM boot before Reset_Handler.